javascript/javascript3/week3/shoppingCart.js

```diff
@@ -0,0 +1,107 @@
+class Product {
+    constructor(name, price) {
+      this.name = name;
+      this.price = price;
+    }
+    convertToCurrency(currency){
+      let price=0;
+      if(currency==="dollars"){
+        price=this.price/6.66;
+      }
+      else if(currency==="rial"){
+        price=this.price*46800;
+      }
+      else if(currency==="euro"){
+        price=this.price/7.5;
+      }else{
+        price=this.price;
+      }
+      return  price;
+    }
+  }
+  
+  class ShoppingCart {
+    constructor() {
+      this.products = [];
+    }
+  
+    addProduct(product) {
+      // Implement functionality here
+      this.products.push(product);
+    }
+  
+    removeProduct(product) {
+      this.products=this.products.filter(function(p){return (p.name!==product.name);});
+      // Implement functionality here
+    }
+  
+    searchProduct(productName) {
+      let foundProducts=this.products.filter(function(p){return p.name===productName;});
+      return foundProducts;
+      // Implement functionality here
+    }
+  
+    getTotal() {
+      let total=0;
+      for(let i=0;i<this.products.length;i++){
+        total+=this.products[i].price;
+      }
+      return total;
+      // Implement functionality here
+    }
+  
+    renderProducts() {
+      let str='';
+      for(let i=0;i<this.products.length;i++){
+        str+='<li>name : ' + this.products[i].name + ' price:' + this.products[i].price + '</li>';
```

Comment:
can be rewritten with string literals `<li>name : ${this.products[i].name} price:` etc. 
This way you don't need to close and reopen your quotes and use `+` every time

Reply:
i heard its not safe when it comes to backend and security, isnt it?

Reply:
I worked with JS / TS for about 3 years, and never saw anyone worrying about it. Googling it also gave me some hits, but from what i can see it's not something you should concern yourself with. So my advice is: use it until somebody else tells you you can't :) It's a very convenient construct.